components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -37,6 +37,10 @@ class ToolbarPresenter(
 
     private var scope: CoroutineScope? = null
 
+    // {Key: WebExtension id, Value: WebExtensionToolbarAction}
+    @VisibleForTesting
+    internal val webExtensionToolbarMap = HashMap<String, WebExtensionToolbarAction>()
```

Comment:
Do we need this map to be here (which makes the presenter stateful) or could this live in the state directly (e.g. we add this as an optional field to the webextension state)

Reply:
The `WebExtensionToolbarAction` is mutable UI state, so I think at least in its current form we should not be adding it to the webextension state. 

We could make it immutable, but then I think we need toolbar API to support updating individual browser actions.

Reply:
Just to clarify because the naming here overlaps. The web extension `BrowserAction` is already in the state, both globally and per tab (to support tab overrides). The `WebExtensionToolbarAction` is just a wrapper for it for displaying purposes in the toolbar, which is why I wouldn't add it to the state. I would prefer for it to be immutable though, but this would require toolbar API.